sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -320,74 +356,39 @@ private static int speculateContentLength(HttpHeaders headers) {
-
-            switch (protocol) {
-                case "https":
-                    try (SSLSocket socket = (SSLSocket) SSL_SOCKET_FACTORY.createSocket(host, port)) {
-                        return doInputOutput(httpRequest, socket);
-                    }
-
-                case "http":
-                    try (Socket socket = new Socket(host, port)) {
-                        return doInputOutput(httpRequest, socket);
-                    }
-
-                default:
-                    throw LOGGER.logThrowableAsWarning(
-                        new ProtocolException("Only HTTP and HTTPS are supported by this client."));
-            }
-        }
 
         /**
          * Calls buildAndSend to send a String representation of the request across the output stream, then calls
          * buildResponse to get an instance of HttpUrlConnectionResponse from the input stream
          *
          * @param httpRequest The HTTP Request being sent
-         * @param socket An instance of the SocketClient
+         * @param bufferedInputStream the input stream from the socket
+         * @param outputStream the output stream from the socket for writing the request
          * @return an instance of Response
          */
-        @SuppressWarnings("deprecation")
-        private static Response<?> doInputOutput(HttpRequest httpRequest, Socket socket) throws IOException {
+        private static Response<?> sendPatchRequest(HttpRequest httpRequest, BufferedInputStream bufferedInputStream,
+            OutputStream outputStream) throws IOException {
             httpRequest.getHeaders().set(HttpHeaderName.HOST, httpRequest.getUrl().getHost());
-            if (!"keep-alive".equalsIgnoreCase(httpRequest.getHeaders().getValue(HttpHeaderName.CONNECTION))) {
-                httpRequest.getHeaders().set(HttpHeaderName.CONNECTION, "close");
-            }
-
-            try (BufferedReader in = new BufferedReader(
-                new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8));
-                OutputStreamWriter out = new OutputStreamWriter(socket.getOutputStream())) {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(bufferedInputStream, StandardCharsets.UTF_8));
+            OutputStreamWriter out = new OutputStreamWriter(outputStream);
 
-                buildAndSend(httpRequest, out);
+            buildAndSend(httpRequest, out);
+            Response<?> response = buildResponse(httpRequest, reader);
 
-                Response<?> response = buildResponse(httpRequest, in);
-                HttpHeader locationHeader = response.getHeaders().get(HttpHeaderName.LOCATION);
-                String redirectLocation = (locationHeader == null) ? null : locationHeader.getValue();
+            HttpHeader locationHeader = response.getHeaders().get(HttpHeaderName.LOCATION);
+            String redirectLocation = (locationHeader == null) ? null : locationHeader.getValue();
```

Comment:
Handling redirect is different in the SocketClient, should add comments to why that is.

Reply:
Didn't specifically look into redirects for this PR, just relayed what we had. 
Let me add a follow up issue for that.
https://github.com/orgs/Azure/projects/579/views/2?pane=issue&itemId=61027721